### PR TITLE
DNN: Upgrade ONNX to Opset 21, Support External Weights, and Fix Implicit Kernel Shape

### LINKED ISSUE #28321: DNN: ONNX Conv import fails when kernel_shape is omitted (spec-compliant models)
### System Information

- OpenCV version: 4.12.0 (opencv-python)
- Backend: CPU
- OS: Windows
- ONNX opset: 21

### Detailed description

When importing a valid ONNX model exported from PyTorch (MobileNetV4-style),
OpenCV DNN fails on Conv nodes that omit the `kernel_shape` attribute.

According to the ONNX specification, `kernel_shape` is optional for Conv and
should be inferred from the convolution weight tensor shape when missing.

OpenCV currently errors out instead of inferring the kernel size.

```
### ERROR MESSAGE
[ERROR] DNN/ONNX: ERROR during processing node [Conv]
kernel_size (or kernel_h and kernel_w) not specified
in cv::dnn::util::getKernelSize
```

### Steps to reproduce

- PyTorch-exported ONNX model (opset 21)
- Multiple Conv nodes omit `kernel_shape` (e.g. node_Conv_48, node_Conv_51, node_Conv_54)
- ONNX checker passes successfully

The exported ONNX model is valid and passes `onnx.checker.check_model`.

###Minimal OpenCV DNN reproducer
```python
import cv2
net = cv2.dnn.readNetFromONNX("mobilenetv4.onnx")
```
### ONNX export script
```python

import torch
import torch.nn as nn

# Minimal MobileNetV4-style model (export-safe)
class MobileNetV4(nn.Module):
    def __init__(self):
        super().__init__()
        self.net = nn.Sequential(
            nn.Conv2d(3, 32, kernel_size=3, stride=2, padding=1, bias=False),
            nn.BatchNorm2d(32),
            nn.ReLU(inplace=True),

            nn.Conv2d(32, 32, kernel_size=3, stride=1, padding=1, groups=32, bias=False),
            nn.BatchNorm2d(32),
            nn.ReLU(inplace=True),

            nn.Conv2d(32, 64, kernel_size=1, bias=False),
            nn.BatchNorm2d(64),
            nn.ReLU(inplace=True),

            nn.AdaptiveAvgPool2d(1),
            nn.Flatten(),
            nn.Linear(64, 1000)
        )

    def forward(self, x):
        return self.net(x)

model = MobileNetV4().eval()

dummy_input = torch.randn(1, 3, 224, 224)

torch.onnx.export(
    model,
    dummy_input,
    "mobilenetv4.onnx",
    opset_version=21,
    input_names=["input"],
    output_names=["output"],
    dynamic_axes=None
)

print("Exported mobilenetv4.onnx successfully")
```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

### PR DESCRIPTION
**Pull request by shahkarnav115-beep**
### Summary
This PR upgrades the internal ONNX Protobuf definitions to **Opset 21 (v1.16.0)** to support modern PyTorch exports (e.g., MobileNetV4) that use **External Data** and omit attributes like `kernel_shape`.

### Resolves
Fixes #28321

### Changes
1.  **Protobuf Definitions:**
    * Updated `onnx.proto` to Opset 21 (v1.16.0).
    * Regenerated `opencv-onnx.pb.cc` and `opencv-onnx.pb.h` using `protoc 3.19.1`.

2.  **Importer Logic (`onnx_importer.cpp`):**
    * Fixed `parseConv`: When `kernel_shape` is missing (allowed in newer Opsets), the importer now correctly infers the kernel size from the weight tensor's shape. This resolves the `kernel_size not specified` error reported in #28321.
    * Added logic to reshape 1D/2D weights for Convolution if needed (Opset 21 optimization).

3.  **Model Loader (`onnx_graph_simplifier.cpp`):**
    * Updated `getMatFromTensor`: Added support for loading weights where `data_location` is `EXTERNAL`. This is required for Opset 21 models which often default to storing weights in external files.

### Testing
* **Unit Test:** Added `Test_ONNX_Importer.Load_External_Weights`.
* **Manual Verification:** Verified against the `mobilenetv4.onnx` reproducer from issue #28321. The model now loads and runs inference successfully.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by shahkarnav115-beep** [comment]
@asmorkalov  @vpisarev  @abhishek-gola

This PR upgrades the ONNX importer to support Opset 21 and fixes the external weights issue (#28321).

I have also submitted the required test data (onnx + bin) to the opencv_extra repo here:
opencv/opencv_extra#1306

Could you please review? Thanks!

**Comment by shahkarnav115-beep** [comment]
Hi, just to clarify the large diff:
this PR updates onnx.proto to support ONNX opset 21 external weights.
As a result, opencv-onnx.pb.h/.cc were regenerated with protoc to keep the generated sources in sync with the schema.
The actual functional changes are in:

-modules/dnn/src/onnx/onnx_importer.cpp
-modules/dnn/src/onnx/onnx_graph_simplifier.cpp
-modules/dnn/misc/onnx/onnx.proto
-test/test_onnx_importer.cpp

**Comment by shahkarnav115-beep** [comment]
closing the PR#28392 to avoid confusion
